### ISSUE TITLE
Cannot use fine-tuned ssd_mobilenet_v2 model with TF Lite on Android

### ISSUE THREAD

**Issue by MichaelW17** [tensorflow/tensorflow#35419]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Win10
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Pixel 2 Simulator in Android Studio with Android 7.1.1
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 1.12.2
- Python version: 3.6.9
- Bazel version (if compiling from source): 0.15.0
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 9.0/7.5
- GPU model and memory: 1070 max-q 8 GB

**Describe the current behavior**
I fine-tuned the ssd_mobilenet_v2 pretrained model from Tensorflow model zoo to detect two classes. Training was done with TF OD API. After converting the model to detect.tflite and put it in the "assets" folder of the [official Android demo](https://github.com/tensorflow/examples/tree/master/lite/examples/object_detection/android) and modified labelmap.txt, upon running I got the following error:
```java
E/AndroidRuntime: FATAL EXCEPTION: inference
    Process: org.tensorflow.lite.examples.detection, PID: 4609
    java.lang.IllegalArgumentException: Internal error: Failed to run on the given Interpreter: tensorflow/lite/kernels/detection_postprocess.cc:404 ValidateBoxes(decoded_boxes, num_boxes) was not true.
    Node number 102 (TFLite_Detection_PostProcess) failed to invoke.
    
        at org.tensorflow.lite.NativeInterpreterWrapper.run(Native Method)
        at org.tensorflow.lite.NativeInterpreterWrapper.run(NativeInterpreterWrapper.java:152)
        at org.tensorflow.lite.Interpreter.runForMultipleInputsOutputs(Interpreter.java:296)
        at org.tensorflow.lite.examples.detection.tflite.TFLiteObjectDetectionAPIModel.recognizeImage(TFLiteObjectDetectionAPIModel.java:193)
        at org.tensorflow.lite.examples.detection.DetectorActivity$2.run(DetectorActivity.java:183)
        at android.os.Handler.handleCallback(Handler.java:751)
        at android.os.Handler.dispatchMessage(Handler.java:95)
        at android.os.Looper.loop(Looper.java:154)
        at android.os.HandlerThread.run(HandlerThread.java:61)
```
However, if I use model.ckpt-0 which is the initial checkpoint, the converted .tflite works fine. So I assume that the convertion is OK. Is it due to the training process?
Please help me figure this out.

**Comment by MichaelW17** [comment]
A walk around that is found by now is:
use the saved checkpoint I need, say, model-ckpt-10000, as the initialization of a new train. Then use the model.ckpt-0 to convert to tflite.

**Comment by srjoglekar246** [comment]
Can you paste the commands you used to convert? If renaming to model.ckpt-0 works for you, there may be some simple bug in the conversion process.

**Comment by hvico** [comment]
Having the same problem here. This used to work, now I do not know what changed in my environment (maybe TFLITE interpreter for Android?) I tried 1.15, 2.0 and 2.1.0.

Error: ValidateBoxes(decoded_boxes, num_boxes) was not true when I try to use the SSD converted model.

Commands I used to convert my checkpoint: python3 object_detection/export_tflite_ssd_graph.py --pipeline_config_path $PIPELINE_PATH --trained_checkpoint_prefix $CHECKPOINT --output_directory $OUTPUT_DIR --add_postprocessing_op true

toco --graph_def_file=/tmp/tflite/tflite_graph.pb --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE --output_file=/tmp/tflite/segm_augm_200k.tflite --input_type=FLOAT --input_arrays=normalized_input_image_tensor --output_arrays='TFLite_Detection_PostProcess','TFLite_Detection_PostProcess:1','TFLite_Detection_PostProcess:2','TFLite_Detection_PostProcess:3' --input_shapes=1,300,300,3  --inference_type=FLOAT --mean_values=128 --std_dev_values=128 --change_concat_input_ranges=false --allow_custom_ops

I dindn't manage to make it work using the workaround MichalW17 provided, I run a training using my pre-trained checkpoint (renamed to 0) with just one step, and then I convert checkpoint 0 file. Then the model loads in Android but it seems it becomes garbage (it acts like the training process was not done).

**Comment by srjoglekar246** [comment]
Can you try without specifying `--mean_values=128 --std_dev_values=128`? Since the inference is in FLOAT, these parameters are not needed. I am not sure what `change_concat_input_ranges` does.

**Comment by hvico** [comment]
Hi! I removed those params, it still crashes with the same error:

2020-01-10 17:43:04.877 22107-22172/org.tensorflow.lite.examples.detection E/AndroidRuntime: FATAL EXCEPTION: inference
    Process: org.tensorflow.lite.examples.detection, PID: 22107
    java.lang.IllegalArgumentException: Internal error: Failed to run on the given Interpreter: tensorflow/lite/kernels/detection_postprocess.cc:404 ValidateBoxes(decoded_boxes, num_boxes) was not true.
    Node number 98 (TFLite_Detection_PostProcess) failed to invoke.

I took this set of params from my own documentation I used a couple of months ago that was working.

Thanks,

**Comment by hvico** [comment]
BTW, this issue seems to be duplicated here: https://github.com/tensorflow/models/issues/7918

**Comment by hvico** [comment]
Update: Downgraded TFLITE to 1.12.0 in Android and now the model loads and works.
Unfortunatelly to do that I had to disable the GPU delegate I was using for improved mobile performance. 
As I explained this procedure was woking months ago (with GPU delegate), as I was using nightly versions of tlite and tflite-gpu. It seems I updated the nightly version recently and now I have this problem.

**Comment by srjoglekar246** [comment]
This is weird. If you don't mind, could you try one of the models you obtained with a previous version of Android, with the Interpreter from the latest version? I am wondering if this is a problem with the conversion process or some TF issues in the latest release.

**Comment by hvico** [comment]
I can confirm my old models (converted some time ago) work properly using the latest version.

**Comment by csmcallister** [comment]
I had the same issue using the `ssd_mobilenet_v2_quantized_300x300_coco` model. I was able to finetune for a couple dozen steps on my data and then export the graph, convert to tf lite, and use the python tf lite interpreter. But when I finetuned on my own data again for ~200,000 steps and repeated the same process, it failed with the `ValidatedBoxed` error message. I'm on Windows 10 using TF version 1.15.0 (through conda as tensorflow-gpu). **Downgrading to tensorflow-gpu=1.13 got it to work.**

**Comment by nrasadi** [comment]
I have also the same issue.
- OS Platform and Distribution: Linux Ubuntu 16.04
- Model: SSD MobileNetV2
- Tensorflow-gpu=2.2 (Installed using conda)

I fine-tuned the model on my own dataset (11 classes). It has been converted successfully but the mentioned error occurs when I call invoke() method.
I am using Python API of TFLite interpreter.

**Code:**
`
tflite_interpreter = tf.compat.v1.lite.Interpreter(model_path=tflite_model_file)
tflite_interpreter.allocate_tensors()
input_details = tflite_interpreter.get_input_details()
output_details = tflite_interpreter.get_output_details()
image = cv2.imread('images.jpeg')
image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
image = cv2.resize(image, (300, 300))
image = np.expand_dims(image, 0)
tensor = tf.compat.v1.convert_to_tensor(np.array(image) / 255. , dtype='float32')
tflite_interpreter.set_tensor(input_details[0]['index'], tensor)
tflite_interpreter.invoke()
`

**Comment by tensorflowbutler** [comment]
Hi There,

 We are checking to see if you still need help on this, as you are using an older version of tensorflow which is officially considered end of life . We recommend that you upgrade to the latest 2.x version and let us know if the issue still persists in newer versions. Please open a new issue for any help you need against 2.x, and we will get you the right help. 

 This issue will be closed automatically 7 days from now. If you still need help with this issue, please provide us with more information.
